### PR TITLE
Add twitch layer

### PR DESCRIPTION
**Pull request by BenediktBroich**
This layer adds support for twitch. You can search for streamers and open the stream in the browser.
If the =erc= layer is enabled you are able to connect to twitch irc and join the streamer channel.
If the =streamlink= layer is enabled the streams are opened with streamlink. (Pull request is open)

### CODE REVIEW

**Comment by BenediktBroich** [comment]
streamlink layer: https://github.com/syl20bnr/spacemacs/pull/14979

**Review by lebensterben — APPROVED** [review]
LGTM

File: layers/+web-services/twitch/packages.el [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+;;; Code:
+
+(defconst twitch-packages
+  '(
+    ;; (erc-twitch :toggle (configuration-layer/package-usedp 'erc))
+    (twitch-api :location (recipe :fetcher github
+                                  :repo "benediktbroich/twitch-api"))
+    (helm-twitch :location (recipe :fetcher github
+                                   :repo "benediktbroich/helm-twitch")
+                 ;; :toggle (configuration-layer/package-usedp 'helm)
```
Comment (lebensterben) [inline layers/+web-services/twitch/packages.el:32]: One issue is this depends on `helm` so you indeed need some "toggle".
Here you can use `:requires`.

File: layers/+web-services/twitch/packages.el [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+;; GNU General Public License for more details.
+;;
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Code:
+
+(defconst twitch-packages
+  '(
+    ;; (erc-twitch :toggle (configuration-layer/package-usedp 'erc))
```
Comment (lebensterben) [inline layers/+web-services/twitch/packages.el:27]:
```suggestion
    ;; (erc-twitch :toggle (configuration-layer/package-usedp 'erc))
```

**Review by lebensterben — CHANGES_REQUESTED** [review]
There's no significant issue with this PR. But there's also no significant feature either.

I'd prefer to see a new layer to include more features/configurations rather than just being a collection of packages.

**Comment by smile13241324** [comment]
Thank you for contributing to spacemacs 💜, I have cherry picked your commit into develop.

**Pull request closed without merging**